src/app/georeferencer/qgsgcplist.cpp

```diff
@@ -30,16 +33,27 @@ QgsGCPList::QgsGCPList( const QgsGCPList &list )
   }
 }
 
-void QgsGCPList::createGCPVectors( QVector<QgsPointXY> &mapCoords, QVector<QgsPointXY> &pixelCoords )
+void QgsGCPList::createGCPVectors( QVector<QgsPointXY> &mapCoords, QVector<QgsPointXY> &pixelCoords, const QString &wktProj )
 {
+  QgsCoordinateReferenceSystem projTarget( wktProj );
+  QgsCoordinateTransform transformer;
   mapCoords   = QVector<QgsPointXY>( size() );
   pixelCoords = QVector<QgsPointXY>( size() );
+  QgsPointXY transCoords;
   for ( int i = 0, j = 0; i < sizeAll(); i++ )
   {
     QgsGeorefDataPoint *pt = at( i );
     if ( pt->isEnabled() )
     {
-      mapCoords[j] = pt->mapCoords();
+      if ( projTarget.isValid() )
+      {
+        transCoords =  QgsCoordinateTransform( pt->crs(), projTarget,
+                                               QgsProject::instance() ).transform( pt->mapCoords() );
```

Comment:
You'll have to wrap the transform call within a try{} and catch exceptions (e.g. https://github.com/qgis/QGIS/blob/master/src/app/locator/qgsinbuiltlocatorfilters.cpp#L1007 )